chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1152 @@
+
+import time
+import struct
+import logging
+from chirp import bitwise, chirp_common, directory, errors, memmap, util
+from chirp.settings import (
+    RadioSetting,
+    RadioSettings,
+    RadioSettingGroup,
+    RadioSettingValueBoolean,
+    RadioSettingValueList,
+    RadioSettingValueInteger,
+    RadioSettingValueString
+)
+
+
+LOG = logging.getLogger(__name__)
+
+COMMAND_ACCEPT = b"\x06"
+CHARSET_HEX = "0123456789ABCDEFabcdef"
+
+
+def _clean_buffer(radio):
+    radio.pipe.reset_input_buffer()
+    radio.pipe.reset_output_buffer()
+    time.sleep(0.002)
+
+    radio.pipe.timeout = 0.005
+    junk = radio.pipe.read(256)
+    radio.pipe.timeout = 1  # 1000ms
+
+    if junk:
+        LOG.debug("Got %i bytes of junk before starting" % len(junk))
+
+
+def _rawrecv(radio, amount):
+    """Raw read from the radio device"""
+    data = bytearray()
+    try:
+        while len(data) < amount:
+            chunk = radio.pipe.read(amount - len(data))
+            if not chunk:
+                LOG.debug("No response from radio")
+                raise errors.RadioNoContactLikelyK1()
+            data.extend(chunk)
+            time.sleep(0.002)  # 2ms delay
+    except Exception:
+        msg = "Generic error reading data from radio; \
+            restart radio and check your cable."
+        raise errors.RadioError(msg)
+
+    if len(data) != amount:
+        LOG.error("Expected %i bytes, got %i" % (amount, len(data)))
+        raise errors.RadioError("Invalid response from the radio")
+
+    return bytes(data)
+
+
+def _rawsend(radio, data):
+    """Raw send to the radio device"""
+    try:
+        radio.pipe.write(data)
+        time.sleep(0.002)
+    except Exception:
+        raise errors.RadioError("Error sending data to radio")
+
+
+def _recv(radio, addr, length):
+    """Get data from the radio """
+    hdr = _rawrecv(radio, 4)  # Read 4 byte header
+    data = _rawrecv(radio, length)  # Read data
+
+    # DEBUG
+    LOG.info("Response:")
+    LOG.debug(util.hexprint(hdr + data))
+
+    c, a, resp_length = struct.unpack(">BHB", hdr)
+    if a != int.from_bytes(addr, byteorder="big") \
+            or resp_length != length or c != ord("W"):
+        LOG.error("Invalid answer for block 0x%04s:" % util.hexprint(addr))
+        LOG.debug("CMD: %s  ADDR: %04x  SIZE: %02x" % (c, a, resp_length))
+        raise errors.RadioError("Unknown response from the radio")
+
+    return data
+
+
+def _enter_programming_mode(radio):
+    _clean_buffer(radio)
+
+    _rawsend(radio, radio._magic)
+    LOG.debug("Sent magic sequence")
+    ack = _rawrecv(radio, 2)
+    LOG.debug("Received magic sequence response")
+    if len(ack) != 2 or ack[1:2] != COMMAND_ACCEPT:
+        if ack:
+            LOG.error("Received: Len=%i Data=%s"
+                      % (len(ack), util.hexprint(ack)))
+        raise errors.RadioError("Radio refused to enter programming mode")
+
+    radio.pipe.reset_input_buffer()
```

Comment:
Same here, please don't do this unless there's some really (really) good and well-tested reasoning.